influxql/ast_test.go

```diff
@@ -1158,6 +1158,57 @@ func Test_fieldsNames(t *testing.T) {
 
 }
 
+func TestSelect_ColumnNames(t *testing.T) {
+	for i, tt := range []struct {
+		stmt    *influxql.SelectStatement
+		columns []string
+	}{
+		{
+			stmt: &influxql.SelectStatement{
+				Fields: influxql.Fields([]*influxql.Field{
+					{Expr: &influxql.VarRef{Val: "value"}},
+				}),
+			},
+			columns: []string{"time", "value"},
+		},
+		{
+			stmt: &influxql.SelectStatement{
+				Fields: influxql.Fields([]*influxql.Field{
+					{Expr: &influxql.VarRef{Val: "value"}},
+					{Expr: &influxql.VarRef{Val: "value"}},
+					{Expr: &influxql.VarRef{Val: "value_1"}},
+				}),
+			},
+			columns: []string{"time", "value", "value_1", "value_1_1"},
+		},
+		{
+			stmt: &influxql.SelectStatement{
+				Fields: influxql.Fields([]*influxql.Field{
+					{Expr: &influxql.VarRef{Val: "value"}},
+					{Expr: &influxql.VarRef{Val: "value_1"}},
+					{Expr: &influxql.VarRef{Val: "value"}},
+				}),
+			},
+			columns: []string{"time", "value", "value_1", "value_2"},
+		},
+		{
+			stmt: &influxql.SelectStatement{
+				Fields: influxql.Fields([]*influxql.Field{
+					{Expr: &influxql.VarRef{Val: "value"}},
+					{Expr: &influxql.VarRef{Val: "total"}, Alias: "value"},
+					{Expr: &influxql.VarRef{Val: "value"}},
+				}),
+			},
+			columns: []string{"time", "value_1", "value", "value_2"},
+		},
+	} {
+		columns := tt.stmt.ColumnNames()
+		if !reflect.DeepEqual(columns, tt.columns) {
+			t.Errorf("%d. expected %s, got %s", i, tt.columns, columns)
```

Comment:
OK, suits me. Personally, I've be converting every test file I touch, but
it can mean some significant changes in some files!

On 22 February 2016 at 13:43, Jonathan A. Sternberg <
notifications@github.com> wrote:

> In influxql/ast_test.go
> <https://github.com/influxdata/influxdb/pull/5742#discussion_r53625443>:
>
> > +			columns: []string{"time", "value", "value_1", "value_2"},
> > +		},
> > +		{
> > +			stmt: &influxql.SelectStatement{
> > +				Fields: influxql.Fields([]*influxql.Field{
> > +					{Expr: &influxql.VarRef{Val: "value"}},
> > +					{Expr: &influxql.VarRef{Val: "total"}, Alias: "value"},
> > +					{Expr: &influxql.VarRef{Val: "value"}},
> > +				}),
> > +			},
> > +			columns: []string{"time", "value_1", "value", "value_2"},
> > +		},
> > +	} {
> > +		columns := tt.stmt.ColumnNames()
> > +		if !reflect.DeepEqual(columns, tt.columns) {
> > +			t.Errorf("%d. expected %s, got %s", i, tt.columns, columns)
>
> I'm ok with switching to that style, but I would like to do it as part of
> a separate commit for that purpose. I don't want to mix coding style within
> a single test file if it is avoidable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdata/influxdb/pull/5742/files#r53625443>.
>



-- 
*Edd Robinson*
InfluxDB Engineer
▼▴

*InfluxData.com <http://influxdata.com/>*

Github <http://www.github.com/e-dard> / LinkedIn
<https://www.linkedin.com/in/eddrobinson> / Personal <http://eddrobinson.io>